File: go/71_SimplifyPath_test.go
package main

import (
    "github.com/stretchr/testify/assert"
    "testing"
)

func TestSimplifyPath(t *testing.T) {
    assert.Equal(t,
        "/home",
        simplifyPath("/home/"),
    )
    assert.Equal(t,
        "/",
        simplifyPath("/../"),
    )
    assert.Equal(t,
        "/home/foo",
        simplifyPath("/home//foo/"),
    )
    assert.Equal(t,
        "/c",
        simplifyPath("/a/./b/../../c/"),
    )
    assert.Equal(t,
        "/c",
        simplifyPath("/a/../../b/../c//.//"),
    )
    assert.Equal(t,
        "/a/b/c",
        simplifyPath("/a//b////c/d//././/.."),
    )
}

File: python/71_SimplifyPath_test.py
import unittest


class TestSolution(unittest.TestCase):

    def test_LengthOfLongestSubstring(self):
        Solution = __import__('71_SimplifyPath').Solution
        self.assertEqual(
            "/home",
            Solution().simplifyPath("/home/"),
        )
        self.assertEqual(
            "/",
            Solution().simplifyPath("/../"),
        )
        self.assertEqual(
            "/home/foo",
            Solution().simplifyPath("/home//foo/"),
        )
        self.assertEqual(
            "/c",
            Solution().simplifyPath("/a/./b/../../c/"),
        )
        self.assertEqual(
            "/c",
            Solution().simplifyPath("/a/../../b/../c//.//"),
        )
        self.assertEqual(
            "/a/b/c",
            Solution().simplifyPath("/a//b////c/d//././/.."),
        )
